### PR TITLE
Update to use Vector and Refs in appropriate files

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/sysinfo.jl [hunk 1]
```diff
@@ -130,25 +130,25 @@ function cpu_summary(io::IO=STDOUT, cpu::Array{CPUinfo}=cpu_info())
...
-    uptime_ = Array{Float64}(1)
+    uptime_ = Ref{Float64}()
     Base.uv_error("uv_uptime",ccall(:uv_uptime, Int32, (Ptr{Float64},), uptime_))
-    return uptime_[1]
+    return uptime_[]
 end
 
 function loadavg()
-    loadavg_ = Array{Float64}(3)
+    loadavg_ = Vector{Float64}(3)
```
Comment (musm) [inline base/sysinfo.jl]: This is fine or do Vector arguments need to be wrapped in Refs?

Comment (yuyichao) [inline base/sysinfo.jl]: What needs to be wrapped in `Ref`?

File: base/sysinfo.jl [hunk 2]
```diff
@@ -117,7 +117,7 @@ function _cpu_summary(io::IO, cpu::Array{CPUinfo}, i, j)
     println(io)
 end
 
-function cpu_summary(io::IO=STDOUT, cpu::Array{CPUinfo}=cpu_info())
+function cpu_summary(io::IO=STDOUT, cpu::Vector{CPUinfo}=cpu_info())
```
Comment (yuyichao) [inline base/sysinfo.jl]: This is in priciple wrong but might not matter.

Comment (TotalVerb) [inline base/sysinfo.jl]: `Array` seems like a wrong argument specialization; I would think that perhaps either `AbstractArray{CPUinfo}`, `AbstractVector{CPUinfo}`, `Vector{CPUinfo}`, or `Any` is better.

Comment (musm) [inline base/sysinfo.jl]: I will just leave this as it was

Comment (TotalVerb) [inline base/sysinfo.jl]: It should still be fixed, maybe in a different PR. It makes no sense to allow `Matrix{CPUinfo}` but not `typeof(view(CPUinfo[], 1:0))`.

Comment (musm) [inline base/sysinfo.jl]: agreed, but let's leave it to a different PR as it seems outside the scope of this one :)

File: base/sysinfo.jl (hunk 1 above)
Comment (JeffBezanson) [inline base/sysinfo.jl]: No Ref needed here.
